harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,551 @@
+import argparse
+import json
+import subprocess
+import sys
+from multiprocessing.sharedctypes import Value
+from pathlib import Path
+from typing import Callable, Dict, Union
+
+from termcolor import cprint
+
+import determined
+from determined.common import yaml
+from determined.common.declarative_argparse import Arg, ArgGroup, Cmd, Group
+from determined.common.util import safe_load_yaml_with_exceptions
+from determined.deploy.gke.constants import defaults
+
+
+def make_spec(task_container_defaults: dict, key: str) -> dict:
+    pod_spec = task_container_defaults.get(key)
+    if not pod_spec:
+        pod_spec = {"apiVersion": "v1", "kind": "Pod"}
+    if not pod_spec.get("spec"):
+        pod_spec["spec"] = {}
+    return pod_spec
+
+
+def validate_accelerator_type(s: str) -> None:
+    json_value = subprocess.check_output(
+        ["gcloud", "compute", "accelerator-types", "list", "--format=json(name)"]
+    )
+    json_names = json.loads(json_value)
+    valid_accelerator_types = {accelerator["name"] for accelerator in json_names}
+
+    if s not in valid_accelerator_types:
+        raise argparse.ArgumentTypeError(
```

Comment:
I think `ArgumentError` might be more appropriate here: `ArgumentTypeError` is supposed to be for data types such as date vs int etc.
accelerator-type seems to just be a "value" in that sense.